expression/builtin_vectorized.go

```diff
@@ -103,6 +103,103 @@ func (r *localSliceBuffer) put(buf *chunk.Column) {
 
 type vecRowConverter struct {
 	builtinFunc
+
+	// fields for converting vectorized evaluation to row-based evaluation.
+	buf *chunk.Column
+	sel []int
+}
+
+func newVecRowConverter(underlying builtinFunc) *vecRowConverter {
+	return &vecRowConverter{underlying, nil, nil}
+}
+
+func (c *vecRowConverter) vecEvalRow(evalType types.EvalType, row chunk.Row) (err error) {
+	if c.sel == nil {
+		c.sel = make([]int, 1)
+	}
+	c.sel[0] = row.Idx()
```

Comment:
Should it be `0`? Because this chunk only contains 1 row.

Reply:
No, the row is iterating on the column which actually stores data and may have multiple rows, so the index of this row is not always zero.